### PR TITLE
fix #5826; swap node-sass for dart-sass

### LINKED ISSUE #5826: Replace node-sass
Node-sass is a pain to install for many.  Took me an half an hour of Google, downloading a bunch of MS Visual Studio/C++ stuff that I'll likely only use for node-sass, then re-reinstalling node-sass.  It was great tech 4 years ago, but now feels to be too much when other alternatives are available.

 #### [css-in-js ](https://css-tricks.com/bridging-the-gap-between-css-and-javascript-css-in-js/)
Worth looking into for new projects with some unicorn devs, [many options available](https://blog.bitsrc.io/9-css-in-js-libraries-you-should-know-in-2018-25afb4025b9b), but perhaps too much rewriting for a legacy project like this.

#### [Dart-Sass](https://sass-lang.com/dart-sass)
The reference SASS (RIP Ruby).  Can be used from CLI as binary, or [complied as JS](https://www.npmjs.com/package/dart-sass), which could be a bit slower than binary, but might be fast enough.

#### PostCSS
Seems switching to PostCSS [requires little extra coding](https://medium.com/@im.simonecorsi/moving-from-sass-to-postcss-why-what-and-how-f68b1bc760dc), considering you'll have to sweep though to remove baked-in prefixes & let [Autoprefixer](https://github.com/postcss/autoprefixer) handle that chore. [PostCSS SASS](https://github.com/jonathantneal/postcss-sass) will load Dart SASS if you want less friction, but Dart SASS might have a few differences with libsass.  Perhaps adding some [new CSS tech](https://css-tricks.com/bridging-the-gap-between-css-and-javascript-css-modules-postcss-and-the-future-of-css/#article-header-id-1) will benefit Video.JS also.  Popular enough & used in many large projects.

Most of the projects listed are recently updated.  I vote PostCSS, since it is fast, JS-only to avoid installing extra headaches, reduces the transition costs.

Thoughts?

### PR DESCRIPTION
**Pull request by tomByrer**
## Description
Per #5826, node-sass is difficult to install sometimes.  Dart-sass is the new 'standard'.

## Specific Changes proposed
To get dart-sass to work, I had to [fix the path to node_modules for `@import`](https://github.com/sass/dart-sass/issues/615) (which is backward compatible with node-sass) & [remove the --watch folder](https://github.com/sass/dart-sass/issues/616).

## Testing

Works in Win10 & OSX Sierra, both using VSCode-git-bash & npm.  Direct `npm run build:css` & `npm start` works, as does tests in Firefox & Safari.  Compiling seems to be ~5-10% slower than node-sass, but we're only talking milliseconds.

I haven't tested editing an scss file & see if the `--watch` works (I assume runs only for the main entry & `@import`ed files).

### Thoughts

I have some reservations.  Dart-sass is clearly a [1-woman show](https://github.com/sass/dart-sass/graphs/contributors).  Perhaps the spec changes (as few a there are now) is handled by a team, she writes 99% of the code.  Amazing work, but I question long-term maintenance.  Documentation has not been updated, with references to ruby-sass which is about to mothballed in weeks.

While LibSass may never be 100% reference, it became the de-facto standard.  Node-sass has some minor behavior differences.

So, up to you guys if you want to make the switch; there are pros & cons.

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [X] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,6 +108,7 @@
     "conventional-changelog-cli": "^2.0.11",
     "conventional-changelog-videojs": "^3.0.0",
     "cross-env": "^5.2.0",
+    "dart-sass": "^1.17.2",
```
[the remark this answers is not in the document]
Comment (tomByrer) [reply package.json]: Seems to be the exact same code, but double-listed in npm under different project names.  Note the version, 'last publish', GitHub stats, etc:
https://www.npmjs.com/package/sass
https://www.npmjs.com/package/dart-sass

I chose `dart-sass` to be more clear about the origin.  Both seemed to work the same when I was doing some wiring.

**Comment by tomByrer** [comment]
Any luck in your findings @gkatsev ?

**Comment by gkatsev** [comment]
I haven't had much time to look at it further.

Also, given that all the documentation says to use the `sass` module, I'd probably prefer that, even if they are otherwise equivalent.

**Comment by BrandonOCasey** [comment]
I did some testing with this today:

Over 20 runs of `npm run build:css` with `dart-sass` it took: `3.44865s`
Over 20 runs of `npm run build:css` with `node-sass` it took: `2.8767s`
So a difference of `~16.5%` or `0.57195s` between the two

From a quick visual test on safari/chrome/firefox everything looks the same, but we probably want to test a lot of specifics since the diff has a lot of non-whitespace changes.

Here is a diff between the two that I did (where node-sass generated css, is being changed to dart-sass css)
https://github.com/videojs/video.js/commit/eb852c4acc144c827d210a89a0a1359e8742c6e9?w=1

**Comment by gkatsev** [comment]
Thanks @BrandonOCasey! Looking a bit more carefully, some of the non-whitespace changes are just a change in the ordering of the selectors, for example:
```css
.vjs-icon-play, .video-js .vjs-big-play-button .vjs-icon-placeholder:before, .video-js .vjs-play-control .vjs-icon-placeholder {}
```
was changed to:
```css
.vjs-icon-play, .video-js .vjs-play-control .vjs-icon-placeholder, .video-js .vjs-big-play-button .vjs-icon-placeholder:before {}
```

The difference is that before, `vjs-big-play-button` was first with `vjs-play-control` second but then they swapped.

@BrandonOCasey would you be able to run something like this replace on the output: `s/\n}/}/`? Basically, delete the new line before closing braces, that should minimize the changeset.

**Comment by BrandonOCasey** [comment]
yeah let me see what I can do, I will re-post the diff after fixing some whitespace issues.

**Comment by BrandonOCasey** [comment]
A bit better: https://github.com/videojs/video.js/pull/5903

**Comment by gkatsev** [comment]
Yeah, looks like it's mostly switching around the selectors or putting them on one line. We'll want to double check that it's true for all the changes, though. But seems like this can be made in a minor update for sure.

**Comment by BrandonOCasey** [comment]
OK, I checked all the changes, and found three differences that I made comments on in that pull request. Seems like rounding is a bit different.

**Comment by gkatsev** [comment]
Hiding whitespace changes makes that other PR a lot more manageable.
I don't think that the precision change matters enough to warrant further investigation. Going from 15 3's to 10 3's is pretty much still just a bunch of 3s.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrandonOCasey) [reply package.json]: Since the [docs for dart-sass](https://sass-lang.com/dart-sass) specify the `sass` package, I think we should go with that.

**Comment by gkatsev** [comment]
Also, we probably would want to turn off sourcemaps.

**Comment by gkatsev** [comment]
So, what's I think there are a couple of things left here:
- [ ] switch to the `sass` package
- [ ] disable sourcemaps
- [ ] rebase against master and fix package-lock conflicts

**Comment by gkatsev** [comment]
@tomByrer would you be able to finish the tasks outlines in my [last comment](https://github.com/videojs/video.js/pull/5841#issuecomment-480006201)?

**Comment by gkatsev** [comment]
I've created a new PR ( #6055 ) which dones the above tasks.

**Pull request closed without merging**